networkx/algorithms/hierarchy_coordinates.py

```diff
@@ -0,0 +1,1107 @@
+"""
+========================
+Hierarchy Coordinates
+========================
+Functions to calculate hierarchy coordinates for directed networks.
+
+Hierarchy coordinates consist of treeness, feedforwardness, and orderability, which collectively create a
+'hierarchy morphospace' classifying all network structures. In loose terms, treeness describes how consistently
+nodes higher on the hierarchy (maximal nodes) branch into more nodes, while feedforwardness considers the extent
+information passes vertically (vs horizontally) within a hierarchy, and orderability considers what fraction of the
+network is vertically hierarchical.
+
+Originally from [1]_, adapted implementation accommodates weighted network. For details, see _[2]
+
+.. [1] "On the origins of hierarchy in complex networks."
+ Corominas-Murtra, Bernat, Joaquín Goñi, Ricard V. Solé, and Carlos Rodríguez-Caso,
+ Proceedings of the National Academy of Sciences 110, no. 33 (2013)
+
+.. [2] Appendix of 2013 paper: https://www.pnas.org/highwire/filestream/613265/field_highwire_adjunct_files/0/sapp.pdf
+
+"""
+
+import networkx as nx
+from networkx.utils import not_implemented_for
+import copy
+
+# from sklearn.preprocessing import normalize
+
+__all__ = [
+    "node_weighted_condense",
+    "weight_nodes_by_condensation",
+    "max_min_layers",
+    "leaf_removal",
+    "recursive_leaf_removal",
+    "orderability",
+    "feedforwardness",
+    "graph_entropy",
+    "infographic_graph_entropy",
+    "treeness",
+    "hierarchy_coordinates",
+]
+
+
+# Hierarchy Coordinate Functions
+def node_weighted_condense(A, num_thresholds=8, threshold_distribution=None):
+    """Creates a series of networkx graphs based on generated edge weight thresholds, and returns their node weighted
```

Comment:
Can you make the doc_strings start with a single-line description. This is used as text to go with links to this function in the documentation.  Keep it short.

Then follow up with a paragraph that actually says more completely what the function does.  Try to keep all doc lines around 80-chars or less.

You need to define what "node weighted condense and original graphs" means.